src/operators/publishReplay.ts

```diff
@@ -7,6 +7,8 @@ import { UnaryFunction } from '../interfaces';
 
 export function publishReplay<T>(bufferSize: number = Number.POSITIVE_INFINITY,
```

Comment:
I think we'll need some function signature overloads for this so we don't break compatability for people expecting `ConnectableObservable<T>` to be the only return type... 

See what `publish` is doing here for a better idea: https://github.com/ReactiveX/rxjs/blob/accbcd0c5f9fd5976be3f491d454c4a61f699c4b/src/operators/publish.ts#L6-L8